File: fizzBuzz.js
/*FizzBuzz in JS
Print every number up to the var topNum.
However, if a number is divisible by 3, print "Fizz", if divisible by 5 print "Buzz", or if divisible by 15 print "FizzBuzz".
*/

function fizzBuzz(topNum) {
	for (i = 1; i < topNum; i++)
		//Find if a num is divisble by 15 first. Otherwise, every "FizzBuzz" will just be "Fizz"
		if (i % 15 == 0 )	
		    console.log("FizzBuzz");
		//Due to needing to find things divisble by 15 first, I like to see if a number is divisble by 5 next. However, you could easily switch the order of divisble by 3 or 5.
		else if (i % 5 == 0) {
			console.log("Buzz");
		}
		//Divisble by 3
		else if (i % 3 == 0) {
			console.log("Fizz");
		}
		//Print all numbers that are not divisble by 3, 5, or 15.
		else if (i > 0) {
			console.log(i);
		}
}

fizzBuzz(100);

File: fizzbuzz.rb
=begin
This is Fizzbuzz in Ruby.
The rules of Fizzbuzz are:
-Print all numbers from 1 until the var topNum unless the number matches one of the conditions below
-if the number is divisble by 3 print Fizz
-if the number is divisble by 5 print uzz
-if the number is divisble by 3 and 5 print fizzbuzz
rescue 

=end => e
  
end

def fizzbuzz(topNum)
  arr = []
  (1..topNum).each do |n|
    #look for numbers divisble by 15 first, otherwise fizzbuzz will never print.
    if ((n % 3 == 0) && (n % 5 == 0))
      puts "FizzBuzz"
    elsif (n % 3 == 0)
      puts "Fizz"
    elsif (n % 5 == 0)
      puts "Buzz"
    #print the number if the number is not divisible by
    else
      puts n
    end
  end
  return arr
end

fizzbuzz(100)